packages/core/src/defer/instructions.ts

```diff
@@ -1444,11 +1449,12 @@ export function triggerDeferBlock(lView: LView, tNode: TNode) {
   if (!shouldTriggerDeferBlock(injector, tDetails)) return;
 
   let registry: DeferBlockRegistry | null = null;
-  if (isIncrementalHydrationEnabled(injector)) {
+  if (isIncrementalHydrationEnabled(injector) && lDetails[SSR_UNIQUE_ID] !== null) {
     registry = injector.get(DEFER_BLOCK_REGISTRY);
+    registry.invokeCleanupFns(lDetails[SSR_UNIQUE_ID]);
```

Comment:
nit: minor refactoring to avoid extra property reads:
```suggestion
  const ssrUniqueId = lDetails[SSR_UNIQUE_ID];
  if (isIncrementalHydrationEnabled(injector) && ssrUniqueId !== null) {
    registry = injector.get(DEFER_BLOCK_REGISTRY);
    registry.invokeCleanupFns(ssrUniqueId);
```
I'm also wondering if we should rename `registry.invokeCleanupFns` to `registry.invokeTriggerCleanupFns` (so that it's more aligned with the `invokeAllTriggerCleanupFns` function, which we can also rename to `invokeTriggerCleanupFns`)?